clpy/core/include/clpy/carray.clh

```diff
@@ -513,3 +514,77 @@ static void atomicAdd(__global float* x, float y)
   } while ( new_value.u != old_value.u );
 }
 // TODO(yoriyuki.kitta): Add another types implementation
+
+#ifdef __ULTIMA
+__attribute__((annotate("clpy_no_mangle"))) static half convert_float_to_half(float x);
+#else
+static half convert_float_to_half(float f) {
```

Comment:
Specify which codes are from FP16 library or not. Only codes in this else-endif block?

Reply:
It's better if all codes from other projects are separated into an other file and write these license
 its header, and #include it.

Reply:
I did it.